stock_split_picking/wizards/stock_picking_split.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2019 Camptocamp SA - Alexandre Fayolle
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, models, fields
+from odoo.exceptions import ValidationError
+from odoo.addons import decimal_precision as dp
+from odoo.tools import float_compare, float_is_zero
+
+
+class StockPickingSplit(models.TransientModel):
+    _name = 'stock.picking.split'
+
+    picking_id = fields.Many2one(
+        'stock.picking',
+        string='Picking',
+        required=True,
+        default=lambda s: s.env.context.get('active_id', False)
+    )
+    line_ids = fields.One2many(
+        'stock.picking.split.line',
+        'split_id',
+        string='Split Operations',
+    )
+
+    @api.model
+    def create(self, vals):
+        rec = super(StockPickingSplit, self).create(vals)
+        rec._generate_lines()
+        return rec
+
+    @api.multi
+    def process(self):
+        self.ensure_one()
+        uom_dp = self.env['decimal.precision'].precision_get(
+            'Product Unit of Measure')
+        picking = self.picking_id
+        reserve = False
+        if picking.pack_operation_ids:
+            reserve = True
+            picking.do_unreserve()
+        split_picking = picking.copy(
+            {
+                'name': '/',
+                'move_lines': [],
+                'pack_operation_ids': [],
+            }
+        )
+        # Handle the case where there are several stock.move with the same
+        # product+UoM. We summed the qtys for all these moves in the wizard,
+        # and now we need to spread back the qtys among the different moves
+        # with the same product+UoM.
+        product_split = {}
+        for line in self.line_ids:
+            key = (line.product_id, line.product_uom_id,
+                   line.restrict_lot_id, line.full_split)
+            product_split[key] = line.split_qty
+        new_move_ids = []
+        for move in picking.move_lines:
+            if move.state in ('cancel', 'done'):
+                continue
+            # key = (Product, UoM, Lot, Full split)
+            key = (move.product_id, move.product_uom,
```

Comment:
Another improvement should be (Roadmap ?) putting code to get that key in a method to easy inheritance.